### PR TITLE
Add support for @experimental_disableErrorPropagation

### PR DESCRIPTION
**Pull request by martinbonnin**
This pull request adds support for `@experimental_disableErrorPropagation` to disable error propagation for error aware clients:

```graphql
type User {
  email: String!
  name: String!
}

query GiveMeAllThePartialData @experimental_disableErrorPropagation {
  user {
    email #this field can fail without blowing up the whole user
    name # this one too 🙌
  }
}
```

Fixes https://github.com/graphql-java/graphql-java/issues/3756

See https://github.com/graphql/nullability-wg/discussions/85
See https://github.com/graphql/graphql-spec/pull/1050
See https://github.com/graphql/graphql-js/pull/4348

### CODE REVIEW

File: src/test/groovy/graphql/execution/NonNullableFieldValidatorTest.groovy [hunk 1]
```diff
@@ -7,7 +7,9 @@ import static graphql.schema.GraphQLNonNull.nonNull
 
 class NonNullableFieldValidatorTest extends Specification {
 
-    ExecutionContext context = Mock(ExecutionContext)
+    ExecutionContext context = Mock(ExecutionContext) {
+        propagateErrors() >> true
+    }
 
     def parameters = Mock(ExecutionStrategyParameters) {
         getPath() >> ResultPath.rootPath()
```
Comment (bbakerman) [inline src/test/groovy/graphql/execution/NonNullableFieldValidatorTest.groovy:15]: Can we please get two types of tests in place.

A unit test showing NonNullableFieldValidator working (not throwing exceptions say)

And a end to end integration test with the query directives in place and a DF returning null on a non null type

File: src/main/java/graphql/Enums.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package graphql;
+
+import graphql.language.Description;
+import graphql.language.EnumTypeDefinition;
+import graphql.language.EnumValueDefinition;
+
+/**
+ * The enums that are understood by graphql-java
+ */
+public class Enums {
```
Comment (bbakerman) [inline src/main/java/graphql/Enums.java:10]: Can add @PublicApi to this please

File: src/main/java/graphql/Enums.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    public static final String ON_ERROR = "OnError";
+    public static final String ON_ERROR_PROPAGATE = "PROPAGATE";
+    public static final String ON_ERROR_NULL = "NULL";
```
Comment (bbakerman) [inline src/main/java/graphql/Enums.java:13]: I don't think the string values need to be public per se

File: src/main/java/graphql/execution/Execution.java [hunk 4]
```diff
@@ -1,13 +1,7 @@
...
 
 
-import graphql.ExecutionInput;
-import graphql.ExecutionResult;
-import graphql.ExecutionResultImpl;
-import graphql.ExperimentalApi;
-import graphql.GraphQLContext;
-import graphql.GraphQLError;
-import graphql.Internal;
+import graphql.*;
```
Comment (bbakerman) [inline src/main/java/graphql/execution/Execution.java:4]: we avoid * imports

File: src/main/java/graphql/execution/ExecutionContext.java [hunk 5]
```diff
@@ -3,11 +3,7 @@
 
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
-import graphql.ExecutionInput;
-import graphql.GraphQLContext;
-import graphql.GraphQLError;
-import graphql.Internal;
-import graphql.PublicApi;
+import graphql.*;
```
Comment (bbakerman) [inline src/main/java/graphql/execution/ExecutionContext.java:6]: we avoid * imports

File: src/main/java/graphql/execution/ExecutionContext.java [hunk 6]
```diff
@@ -170,6 +168,9 @@ public ValueUnboxer getValueUnboxer() {
         return valueUnboxer;
     }
 
+    @ExperimentalApi
+    public boolean propagateErrors() { return propagateErrors; }
```
Comment (bbakerman) [inline src/main/java/graphql/execution/ExecutionContext.java:172]: Nice use of @ExperimentalApi - can we have a javadoc comment in it please

File: src/main/java/graphql/schema/idl/SchemaGenerator.java [hunk 7]
```diff
@@ -183,8 +185,12 @@ public boolean isUseAppliedDirectivesOnly() {
             return useAppliedDirectivesOnly;
         }
 
+        public boolean getAddOnErrorDirective() {
+            return addOnErrorDirective;
+        }
```
Comment (bbakerman) [inline src/main/java/graphql/schema/idl/SchemaGenerator.java:190]: should be `isAddOnErrorDirective` in line with the other booleans in this Options class

File: src/main/java/graphql/schema/idl/SchemaGenerator.java [hunk 8]
```diff
@@ -183,8 +185,12 @@ public boolean isUseAppliedDirectivesOnly() {
...
+
         public static Options defaultOptions() {
-            return new Options(true, true, false);
+            return new Options(true, true, false, false);
```
Comment (bbakerman) [inline src/main/java/graphql/schema/idl/SchemaGenerator.java:193]: Should this default to true - if we are going to add this directive I think it should be on by default (so it can be experimentally used) and if some one wants to opt out, then can

Comment (bbakerman) [reply src/main/java/graphql/schema/idl/SchemaGenerator.java:193]: SchemaGenerator is the way MOST schemas are built btw

File: src/main/java/graphql/Directives.java [hunk 9]
```diff
@@ -1,23 +1,13 @@
 package graphql;
 
 
-import graphql.language.BooleanValue;
-import graphql.language.Description;
-import graphql.language.DirectiveDefinition;
-import graphql.language.StringValue;
+import graphql.language.*;
```
Comment (bbakerman) [inline src/main/java/graphql/Directives.java:4]: we avoid * imports

File: src/main/java/graphql/Enums.java (hunk 3 above)
Comment (bbakerman) [reply src/main/java/graphql/Enums.java:13]: I am worried about the `NULL`  name here - its an enum value hence it can be

```
@errorHandling(onError : NULL)` or `@errorHandling(onError : PROPOGATE)
```

`null` is a keyword in graphql language - I think this should be 

* `ALLOW_NULL`

or perhaps the two should be

* `PROPOGATE_NULL`
* `PROPOGATE_ERROR`

which says more about whats to happen

Comment (bbakerman) [reply src/main/java/graphql/Enums.java:13]: yeah

```
def sdl = """
            type Query {
                f(arg1 : AnEnum arg2 : AnEnum) : String
            }
            
            enum AnEnum {
                NULL
                PROPOGATE
            }    
        """

        DataFetcher df = {DataFetchingEnvironment env ->
            println "arg1=" + env.getArgument("arg1")
            println "arg2=" + env.getArgument("arg2")
            return "ok"
        }

        def graphQL = TestUtil.graphQL(sdl, [Query: [f: df]]).build()

        when:
        def er = graphQL.execute("""query q { f(arg1 : NULL arg2 : null) }""")

        then:
        er.errors.isEmpty()
```

reports

```
arg1=NULL. (A string "NULL")
arg2=null
```

I think this is very confusing

File: src/test/groovy/graphql/execution/NonNullableFieldValidatorTest.groovy (hunk 1 above)
Comment (martinbonnin) [reply src/test/groovy/graphql/execution/NonNullableFieldValidatorTest.groovy:11]: Tests added, they found [an issue](https://github.com/graphql-java/graphql-java/pull/3772/commits/eb00f570ac12b8bf06eea43d0344d12b5305516d#diff-d839ec58e4dfc74c0cc52918ee6471da852cfcfdb5422422dc2e5561993470a9R61) already if the resolver returned `null` and did not throw

File: src/main/java/graphql/execution/Execution.java (hunk 4 above)
Comment (martinbonnin) [reply src/main/java/graphql/execution/Execution.java]: Reverted! Any objection to checking in `.idea/codeStyles` to avoid IntelliJ adding those automatically?

File: src/main/java/graphql/execution/ExecutionContext.java (hunk 6 above)
Comment (martinbonnin) [reply src/main/java/graphql/execution/ExecutionContext.java:182]: ✅ let me know what you think

File: src/main/java/graphql/Directives.java [hunk 10]
```diff
@@ -226,6 +247,23 @@ public class Directives {
...
+            .description("This directive controls how to handle errors.")
+            .argument(newArgument()
+                    .name("onError")
+                    .type(nonNull(GraphQLEnumType.newEnum()
+                            .name(Enums.ON_ERROR)
+                            .value(Enums.ON_ERROR_PROPAGATE)
+                            .value(Enums.ON_ERROR_NULL)
+                            .build()))
+                    .defaultValueProgrammatic(Enums.ON_ERROR_PROPAGATE)
+                    .description("The URL that specifies the behaviour of this scalar."))
```
Comment (dondonz) [inline src/main/java/graphql/Directives.java:262]: I think this description is a copy paste from `specifiedBy`, could you update this?

**Comment by bbakerman** [comment]
The graphql-java team @andimarek and @dondonz discussed this today.

We decided that we would accept this feature with the idea that is very much opt in.

So rather than use src/main/java/graphql/schema/idl/SchemaGenerator.java and its Options to "auto put" the directive in place we would require that people copy the directive into their schema themselves.  if some one was programmatically creating a schema say then they would have to add `Directives.ErrorHandlingDirective` themselves into the schema say.

This is in line with how the experimental @defer is also required to be added to the SDL in text form or programmatically added to the schema say

Also we should adopt the same "opt in" mechanism as defer uses eg `graphql.ExperimentalApi#ENABLE_INCREMENTAL_SUPPORT`

We should create another String constant like this and if its present in the graphql context, then this code is enabled

eg something like this

```
var b = Optional.ofNullable(executionContext.getGraphQLContext())
                        .map(graphqlContext -> graphqlContext.getBoolean(ExperimentalApi.ENABLE_ERROR_PROPAGATION))
                        .orElse(false)
```

We have to think of a name of this "feature" and I dont think "error handling" is a great name because its so general versus this very specific "dont bubble non null errors up" form of error handling.

So this also calls into question the directive name of "errorHandling" - again its a general term but in practice this is actually a very specific thing (non null exception propogration or not )

File: src/main/java/graphql/Directives.java (hunk 10 above)
Comment (martinbonnin) [reply src/main/java/graphql/Directives.java]: Thanks for catching this 🙏 . Fixed [there](https://github.com/graphql-java/graphql-java/pull/3772/commits/41863ea4ca974952b77e8d9b7a41c5b4f1703510#diff-cd15824ddc98b9f759a3469c0ca4df14a54820a0512d62954c8bff8d26c114caR263)

Out of curiosity, what is the reason there is both `DirectiveDefinition` and `GraphQLDirective`? I get that `DirectiveDefinition` is the AST one. But what is the need for `GraphQLDirective` too? Feels like it duplicates most of `DirectiveDefinition`?

**Comment by martinbonnin** [comment]
@bbakerman makes a lot of sense, this is very similar to `@defer` 👍 .

I pushed some changes [here](https://github.com/graphql-java/graphql-java/pull/3772/commits/41863ea4ca974952b77e8d9b7a41c5b4f1703510). Let me know what you think.

Re: naming, it's hard 😅 . I went with `ENABLE_CUSTOM_ERROR_HANDLING` but no strong opinion there. If anyone has an idea, please feel free to share.

File: src/main/java/graphql/Directives.java (hunk 10 above)
Comment (bbakerman) [reply src/main/java/graphql/Directives.java]: If some one was to build a GraphqlSchema programmatically (you dont need to use SDL technically) then this is a runtime ready representation of the directive.  A DirectiveDefinition is an AST element, typically from parsing SDL text.   A GraphQLDirective is similar but represents the runtime version of the directive in a GraphqlSchema

File: src/test/groovy/graphql/execution/NoErrorPropagationTest.groovy [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+        ExecutionInput ei = ExecutionInput.newExecutionInput(query).root(
+                [foo: null]
+        ).graphQLContext([(ExperimentalApi.ENABLE_CUSTOM_ERROR_HANDLING): true])
+                .build()
+
+        def er = graphql.execute(ei)
+
+        then:
+        er.data == null
+        er.errors[0].path.toList() == ["foo"]
```
Comment (bbakerman) [inline src/test/groovy/graphql/execution/NoErrorPropagationTest.groovy:72]: Can we check thats its the specific error message (eg the non null propogation error message) and not just something else

**Comment by bbakerman** [comment]
Thanks for the update and removing the SchemaGenerator code.

I think the final part here is the naming. 

Right now its

```
enum OnError {
              ALLOW_NULL
              PROPAGATE
            }
directive @errorHandling(onError: OnError) on QUERY | MUTATION | SUBSCRIPTION
```

My problem with the name is `errorHandling` and `onError` is very generic - eg it sounds a little bit more like perhaps "all the error handling" on this operation.  Whereas in truth its a very specific bit of error handling - it prevents the propagation of exceptions when a non null type field returns null.

I think we should name this directive more specifically about this "non null type field returning null" behavior.

>  Naming is hard - lets go shopping!!

Here are my suggestions (with a matching directive argument and enum type)

* `onNonNullFieldReturningNull`
* `onNonNullFieldContractFailure`
* `onFieldContractFailure`
* `onContractFailure`

The last one is my favourite name because we could extend it for further contract failures in the future (but probably not)

```
enum NonNullFieldIsNull {
              ALLOW_NULL
              PROPAGATE
            }

directive @onContractFailure(nonNullFieldIsNull: NonNullFieldIsNull) on QUERY | MUTATION | SUBSCRIPTION
```

Thoughts?

@andimarek / @dondonz @martinbonnin

**Comment by martinbonnin** [comment]
Thanks for the feedback! 

Re: naming, my initial reaction is that this directive is indeed a very generic thing. It's about how to represent field errors for the whole operation and is not restricted to non-null positions. For an example, we could have this:

```graphql
enum OnError {
  ALLOW_NULL // coerce to null
  PROPAGATE // nearest nullable parent is null
  THROW // stop processing the operation at the first field error (and save CPU cycles if the client does not handle partial data)
}
```

It's not only about "non-null returning null". It's also about "non-null raising an error" and possibly "any field raises an error". 

I could suggest maybe something like:

```graphql
directive @onFieldError(action: ErrorAction) on ...

enum ErrorAction {
  ALLOW_NULL
  PROPAGATE
  THROW
}
```

Not much different but at least it focuses on field errors.

**Comment by martinbonnin** [comment]
For some more background there, the initial discussions mentioned a single directive `@noBubblesPlz` (half-joking name :) but at least everyone got that). 

This single directive had the advantage of simplicity. The possibility and have the server abort an operation completely when an error is encountered and therefore have a 3rd `THROW` case is what motivated the using an enum.

For the sake of this experimental feature, we can limit the scope and maybe keep things simple with a single `@doNotPropagateErrors` directive? 

```graphql
query GetFoo @doNotPropagateErrors {
  foo
}
```

File: src/main/java/graphql/execution/Execution.java (hunk 4 above)
Comment (dondonz) [reply src/main/java/graphql/execution/Execution.java]: That's a great idea, please do check in a code style file

**Comment by dondonz** [comment]
Naming is truly the hardest thing 😄 

Only speaking for myself here, I would like to see a specific name. 

For example for other GraphQL built-in directives we're more specific with directive names:
* It's `@defer` and `@stream` rather than `@incrementalDelivery` with an enum to toggle
* It's `@skip` and `@include` rather than `@conditionalField` with an enum to toggle

As someone who didn't already know about the specification PR, at first glance I was a little confused by `@errorHandling` as the name. Being a general term I wasn't sure about the intended behaviour straight away, and I was confused because there already is a default for error handling in the spec. Having a specific name makes it clearer the behaviour will be different to the default.

I see on the spec PR, the tentative name of the directive is `@nullOnError`, I think that's better because it's more specific and it's short. Other options on that PR description are also specific and could work: `@disableErrorPropagation`, `@disableNullPropagation`. There may be a set of directives as there are options for different behaviours

I'm mindful you've probably thought long and hard about the name already, I don't want to gatecrash this naming discussion! This is only my personal view. You're the spec champions and you are in a better spot to make a decision.

**Comment by martinbonnin** [comment]
@dondonz makes a lot of sense. 

I'm down with `@nullOnError` 👍 [The simple solution is very often a good solution.](https://en.wikipedia.org/wiki/Occam%27s_razor)

> I'm mindful you've probably thought long and hard about the name already

TBH, I've gone through so many naming cycles that I'm not sure of anything anymore 😄 Your perspective is very refreshing. This does not accomodate the future evolution of adding a 3rd enum to save CPU cycles and stop processing at the first error but WAPGNI (We Aren't Probably Gonna Need It!)

I'll make the changes a bit later today.

File: src/main/java/graphql/execution/Execution.java (hunk 4 above)
Comment (martinbonnin) [reply src/main/java/graphql/execution/Execution.java]: Done [here](https://github.com/graphql-java/graphql-java/pull/3772/commits/6542fe2b3acaa033afcd92c572c5916a65b7d3a7)

**Comment by dondonz** [comment]
FYI I re-ran the pipeline because there was a strange 503 error, it was a flake

**Comment by martinbonnin** [comment]
Latest test run failed with:
```
ExecutorInstrumentationTest > will execute on another thread and stay there without a processing executor FAILED
    Condition not satisfied:

    observer.actions == ["FETCHING on FetchThread", "PROCESSING on FetchThread"]
    |        |       |
    |        |       false
    |        [FETCHING on FetchThread, PROCESSING on Test worker]
    <graphql.execution.instrumentation.threadpools.ExecutorInstrumentationTest$TestingObserver@6248af60 actions=[FETCHING on FetchThread, PROCESSING on Test worker]>
        at graphql.execution.instrumentation.threadpools.ExecutorInstrumentationTest.will execute on another thread and stay there without a processing executor(ExecutorInstrumentationTest.groovy:145)
```

I can't reproduce locally. I just merged master, see if it changes anything.

**Comment by dondonz** [comment]
Thanks for reporting the test, glad it's green now! I noticed in the past week a few more flakes than usual

**Comment by martinbonnin** [comment]
Happy new year everyone 🎉 ! Is there anything else needed here?

**Comment by bbakerman** [comment]
I am happy with the PR but I am still not happy with the naming.

The naming is better than it was but its still to generic for my liking

`@nullOnError` sounds like its some behavior on ANY error but its actually very specific and that is it returns null on a non null contract error and prevents null bubbling.

My personal view is to get this to be more specific - and yes naming is hard

* `@nullOnNonNullError`
* `@nullOnNonNullContractError`

**Comment by martinbonnin** [comment]
@bbakerman my hunch is that we'll need a somewhat short and easily pronounceable name if we want the feature to be more largely adopted but at this point I'm fine with a more descriptive one. 

I made the change in @nullOnNonNullError

**Comment by martinbonnin** [comment]
@bbakerman, @dondonz, I have added the naming question to the [next nullabilty working group this Wednesday](https://github.com/graphql/nullability-wg/blob/main/agendas/2025/01-Jan/29-nullability-wg-january-2025.md) ([PR](https://github.com/graphql/nullability-wg/pull/81)).

It might be a bit early in your part of the world but it'd be cool to have you there!

**Comment by dondonz** [comment]
The meeting time is a little too early for me, but keen to see what you all discuss!

**Comment by martinbonnin** [comment]
Meeting just finished. High level notes are available [here](https://docs.google.com/document/d/1IwWB_JBgqFnKVNnXph1k0j3ZTrMzYR9n8fEaxR11Fj4/edit?tab=t.0). Recording available soon at https://www.youtube.com/@GraphQLFoundation

tldr: we ended up with even more proposals 😅:

* `@noErrorPropagation` 
* `@localErrors`
* `@semanticErrors` 

I personnally like that last one because it hints at [semantic nullability](https://github.com/captbaritone/awesome-semantic-nullability) which might be the long-term goal. 

I also like that it introduces a "semantic error" that is reprensented in a json response with a `null` in the `"data"` section and an associated error in the `"errors"` section. Looking at the response more globally instead of just talking about the "nulls" in `"data"` make sense IMO.

Anyways, let us know what you think and if you want to join next working group, feel free to reach out in `#nullability-wg` in the [GraphQL Discord](https://discord.graphql.org/). We can also schedule an ad-hoc meeting too!

**Comment by andimarek** [comment]
FYI: we decided to put this on hold for now because we were informed that the WG works on a new RFC for this topic.

**Comment by benjie** [comment]
I read `@nullOnNonNullError` and thought you were implying that only coercion error caused by a `null` in a non-null position would result in null; but I realise that's not what was intended - instead you meant "non-null positions may be null when there is an error". Perhaps `@ignoreNonNullOnError` or `@allowNullOnError` or just `@nullOnError` might be a closer description? No matter where an error occurs (a non-nullable position or otherwise), in this mode, the position will become null if an error occurs there. An alternative tact could be: `@disableErrorPropagation`

> FYI: we decided to put this on hold for now because we were informed that the WG works on a new RFC for this topic.

This makes sense:

- one nullability proposal has two wrapper types around a default nullable type: (strictly) Non-Null and Semantic-Non-Null; disabling null bubbling then can be thought of as treating every Non-Null wrapper as if it were a Semantic-Non-Null wrapper instead (and this may be expressed through introspection in this mode)
- the other nullability proposal may try and leverage this same work and only change things at a syntax level, but that feels wrong. Instead it will likely have two wrapper types around a default "semantically non-nullable" type: Nullable and (strictly) Non-Null; disabling null bubbling then would be equivalent to stripping all Non-Null wrappers, leaving only Nullable and the unwrapped type which is now "semantically non-nullable"

If you want to implement the introspection changes at the same time then I think you're probably right to hold off implementing it until the latter solution has the spec changes fully written out :+1: 

If you're happy only implementing the non-introspection features, then both proposals work in the same way (error propagation is disabled - errors (including nulls in non-null positions) never bubble) and there's no need to wait.

**Comment by martinbonnin** [comment]
> we decided to put this on hold for now because we were informed that the WG works on a new RFC for this topic.

For the record, I believe we shouldn't always wait for an RFC to be complete before adding (experimental) features. 

The WG discussion have been going on for a while now and getting something relatively simple out so that [people can toy around with](https://slack-chats.kotlinlang.org/t/23464547/hey-folks-the-topic-of-partial-responses-has-come-up-interna#a67b64ab-26dc-4310-ad13-f37dece4267f) has lots of value IMO. 

That being said, alignment is also good. Let's discuss!

In the meantime, I'll revert to `@nullOnError`. This was our first hunch and I'll trust that the simplicity of it outweights the other drawbacks.

**Comment by andimarek** [comment]
@michaelstaib can you elaborate on your plans here? thanks!

**Comment by martinbonnin** [comment]
I've made a matching pull request there for graphql-js: https://github.com/graphql/graphql-js/pull/4348

You'll notice the naming is a tiny bit different, I kept the enum over there. After some back and forth, I think I keep it overall nicer as it really explicits the different choices. It gives the community different choices but overall, no strong opinion on the naming here.

**Comment by martinbonnin** [comment]
I've synced this PR with https://github.com/graphql/graphql-js/pull/4348

There was a consensus to make the directive experimental by prefixing it with `experimental_` and move forward even though the ultimate solution might look different (probably request parameter or document directive?). In all cases, the `experimental_` prefix should make it relatively clear.

I have also removed the `GraphQLContext` flag to enable the parsing of the directive. This is to remove friction with potential integrators/users. Users can test without having to wait for a release of their framework of choice by "just" bumping the transitive version of `graphql-java`. Also keeps things in sync with graphql-js. Let me know if this is an issue and I'll add it back.

File: .gitignore [hunk 12]
```diff
@@ -1,7 +1,8 @@
 *.iml
 *.ipr
 *.iws
-.idea
+.idea/*
+!.idea/codeStyles
```
Comment (bbakerman) [inline .gitignore:5]: Sorry can we get rid of these.

I dont want "outside opnions" on how we set up our IDEA in tis PR.

Make another PR if this is something we should do

File: .idea/codeStyles/Project.xml [hunk 13]
```diff
@@ -0,0 +1,8 @@
+<component name="ProjectCodeStyleConfiguration">
+  <code_scheme name="Project" version="173">
+    <JavaCodeStyleSettings>
+      <option name="CLASS_COUNT_TO_USE_IMPORT_ON_DEMAND" value="1000" />
+      <option name="NAMES_COUNT_TO_USE_IMPORT_ON_DEMAND" value="1000" />
+    </JavaCodeStyleSettings>
+  </code_scheme>
+</component>
```
Comment (bbakerman) [inline .idea/codeStyles/Project.xml:8]: Please get rid of this.  It might be better but its nothing to do with this PR

**Review by bbakerman — APPROVED** [review]

**Comment by bbakerman** [comment]
We want to make another change so that `@experimental_disableErrorPropagation` is auto added to SDL like all the other directives from graphql-java are.  We want this for consistency with the other directives

BUT this does not have to be this PR (it will break one test here - which is good - TDD)

File: .gitignore (hunk 12 above)
Comment (martinbonnin) [reply .gitignore]: Sure thing! https://github.com/graphql-java/graphql-java/pull/3772/commits/535a620505004c1b92563ec0f7a2fee4f0d00f3a

**Comment by martinbonnin** [comment]
Thanks for the review!

> We want to make another change so that @experimental_disableErrorPropagation is auto added to SDL like all the other directives from graphql-java are. 

Are you talking about [`GraphQLSchema.builder` here](https://github.com/graphql-java/graphql-java/blob/43149e000730eb853e0afac36ae58a6c3ecfbec8/src/main/java/graphql/schema/GraphQLSchema.java#L857)?

FWIW, in graphql-js, the user has to add the directive definition manually for it to be recognized. This how the "opt-in" is done. Of course, it's perfectly fine for different projects to have different conventions, just surfacing that here so we're aligned! 

In all cases, let me know if this PR needs anything else.

**Comment by martinbonnin** [comment]
Thanks all for the reviews! Looking forward to trying it out!
